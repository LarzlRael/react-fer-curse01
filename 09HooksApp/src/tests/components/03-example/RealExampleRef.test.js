const { shallow } = require("enzyme");
const { default: RealExampleRef } = require("../../../componentes/04-useRef/RealExampleRef");

describe('Test in <RealExample />', () => {

    const wrapper = shallow(<RealExampleRef />);

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });

    test('should display the component <RealExampleRef />', () => {

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })


})
