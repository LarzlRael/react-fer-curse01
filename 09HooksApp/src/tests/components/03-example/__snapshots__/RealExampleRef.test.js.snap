// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealExample /> should display correctly 1`] = `
<div>
  <h2>
    RealExampleRef
  </h2>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
